crates/data_privacy/src/redactors/rapidhash_redactor.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! A redactor based on `RapidHash`.
+
+use crate::{DataClass, Redactor};
+use core::fmt::Write;
+use rapidhash::v3::{RapidSecrets, rapidhash_v3_seeded};
+
+/// The length of the redacted output in hex digits.
+pub const REDACTED_LEN: usize = 16;
+
+/// A redactor that replaces the original string with the `RapidHash` hash of the string.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct RapidHashRedactor {
+    secrets: RapidSecrets,
+}
+
+impl RapidHashRedactor {
+    /// Creates a new instance with a custom secret.
+    ///
+    /// The secret must be at least 136 bytes long and at most 256 bytes long, with
+    /// a length of 192 being recommended.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the secret is not within the specified length range.
+    #[must_use]
+    pub fn with_secret(secrets: RapidSecrets) -> Self {
+        Self { secrets }
+    }
+}
+
+impl Redactor for RapidHashRedactor {
+    fn redact(&self, _: &DataClass, value: &str, output: &mut dyn Write) -> core::fmt::Result {
+        let hash = rapidhash_v3_seeded(value.as_bytes(), &self.secrets);
+        let buffer = u64_to_hex_array(hash);
+
+        // SAFETY: The buffer is guaranteed to be valid UTF-8 because it only contains hex digits.
+        write!(output, "{}", unsafe { core::str::from_utf8_unchecked(&buffer) })
+    }
+}
+
+#[inline]
+fn u64_to_hex_array(mut value: u64) -> [u8; 16] {
```

Comment:
This here could probably be a `u64_to_hex_array<const N: usize>() -> [u8; N]`, be moved to the parent `mod.rs`, and then be called from all redactors.